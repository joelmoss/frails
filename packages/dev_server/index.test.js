const serializer = require('jest-serializer-path')
const { createConfig } = require('@frails/core')

const devServer = require('.')

describe('devServer', () => {
  beforeAll(() => {
    expect.addSnapshotSerializer(serializer)
  })

  test('createConfig', () => {
    expect(createConfig(devServer)).toMatchSnapshot()
    expect(JSON.stringify(createConfig(devServer()))).toBe(JSON.stringify(createConfig(devServer)))
  })

  test('disabled', () => {
    expect(createConfig(devServer(false))).toMatchSnapshot()
  })
})
