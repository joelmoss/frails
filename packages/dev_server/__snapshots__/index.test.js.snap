// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`devServer createConfig 1`] = `
Object {
  "devServer": Object {
    "contentBase": "<PROJECT_ROOT>/public/assets",
    "port": 8080,
    "public": "localhost:8080",
    "publicPath": "/assets/",
    "watchOptions": Object {
      "ignored": "**/node_modules/**",
    },
  },
  "entry": Object {},
  "mode": "development",
  "output": Object {
    "path": "<PROJECT_ROOT>/public/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    WebpackFixStyleOnlyEntriesPlugin {
      "apply": [Function],
      "options": Object {
        "extensions": Array [
          "less",
          "scss",
          "css",
          "sass",
          "styl",
        ],
        "ignore": undefined,
        "silent": false,
      },
    },
    Object {},
  ],
  "resolve": Object {
    "alias": Object {
      "lib": "<PROJECT_ROOT>/app/lib",
    },
  },
}
`;

exports[`devServer disabled 1`] = `
Object {
  "entry": Object {},
  "mode": "development",
  "output": Object {
    "path": "<PROJECT_ROOT>/public/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    WebpackFixStyleOnlyEntriesPlugin {
      "apply": [Function],
      "options": Object {
        "extensions": Array [
          "less",
          "scss",
          "css",
          "sass",
          "styl",
        ],
        "ignore": undefined,
        "silent": false,
      },
    },
    Object {},
  ],
  "resolve": Object {
    "alias": Object {
      "lib": "<PROJECT_ROOT>/app/lib",
    },
  },
}
`;
